### PR TITLE
Introduce player handlers for the opening and closing of block containers

### PR DESCRIPTION
**Pull request by DenielWorld**
Added HandleContainerOpen() and HandleContainerClose() methods to the player Handler interface.

### CODE REVIEW

**Comment by Sandertv** [comment]
Could you give a concrete example of when these would be useful?

File: server/player/player.go [hunk 1]
```diff
@@ -1581,7 +1581,17 @@ func (p *Player) OpenBlockContainer(pos cube.Pos) {
...
+	p.handler().HandleContainerOpen(ctx, pos)
+
+	ctx.Continue(func() {
+		p.session().OpenBlockContainer(pos)
+	})
+}
+
+// CloseCurrentBlockContainer handles the closing of a block container by the player at the position passed.
+func (p *Player) CloseCurrentBlockContainer(pos cube.Pos) {
+	p.handler().HandleContainerClose(pos)
```
Comment (Sandertv) [inline server/player/player.go:1594]: This method does nothing by itself. Code in `session/player.go` should probably be moved here or this method name will be inaccurate.

**Comment by T14Raptor** [comment]
@DenielWorld Any update?

**Comment by Sandertv** [comment]
Any update?

**Comment by UnknownOre** [comment]
> Could you give a concrete example of when these would be useful?
this could be useful if we wanted to forbid the player from opening a specific  container

**Comment by JustTalDevelops** [comment]
Closing this because aforementioned functionality can be achieved the same way using a UseOnBlock handler.

**Pull request closed without merging**